src/community/oseo/oseo-service/src/main/java/org/geoserver/opensearch/eo/response/TemplatesProcessor.java

```diff
@@ -158,6 +168,29 @@ protected void addUtilityFunctions(String baseURL, Map<String, Object> model) {
                                 throw new RuntimeException("Failed to encode geometry", e);
                             }
                         });
+        model.put("parseJSON", (TemplateMethodModelEx) this::parseJSON);
+    }
+
+    private String parseJSON(List arguments) {
+        String filePath = arguments.get(0).toString();
+        JSONParser parser = new JSONParser();
+        try {
+            GeoServerDataDirectory geoServerDataDirectory =
+                    GeoServerExtensions.bean(GeoServerDataDirectory.class);
+
+            File file = geoServerDataDirectory.findFile(filePath);
+            if (file == null) {
+                LOGGER.warning("File is outside of data directory");
+                throw new RuntimeException(
+                        "File " + filePath + " is outside of the data directory");
+            }
+
+            return parser.parse(new FileReader(file.getPath())).toString();
+        } catch (Exception e) {
+            LOGGER.warning("Failed to parse JSON file " + e.getLocalizedMessage());
+        }
+        LOGGER.warning("Failed to create a JSON object");
+        return "Failed to create a JSON object";
     }
```

Comment:
A better refactor could be
```suggestion
    private TemplateMethodModelEx parseJSON() {
        return arguments -> loadJSON(arguments.get(0).toString());
    }

    private String loadJSON(String filePath){
        JSONParser parser = new JSONParser();
        try {
            GeoServerDataDirectory geoServerDataDirectory =
                    GeoServerExtensions.bean(GeoServerDataDirectory.class);

            File file = geoServerDataDirectory.findFile(filePath);
            if (file == null) {
                LOGGER.warning("File is outside of data directory");
                throw new RuntimeException(
                        "File " + filePath + " is outside of the data directory");
            }

            return parser.parse(new FileReader(file.getPath())).toString();
        } catch (Exception e) {
            LOGGER.warning("Failed to parse JSON file " + e.getLocalizedMessage());
        }
        LOGGER.warning("Failed to create a JSON object");
        return "Failed to create a JSON object";
    }
```

Reply:
done